test/onnx/exporter/test_small_models_e2e.py

```diff
@@ -263,6 +264,185 @@ def forward(self, x):
         onnx_program = self.export(PrintModule(), (torch.tensor(1),))
         onnx_testing.assert_onnx_program(onnx_program)
 
+    def test_export_with_dynamic_input(self):
+        class Model(torch.nn.Module):
+            def forward(self, x):
+                return x + 1.0
+
+        x = torch.randn(2, 3, 4, dtype=torch.float)
+        dim0 = torch.export.Dim("dim0")
+        onnx_program = self.export(Model(), (x,), dynamic_shapes=({0: dim0},))
+
+        # different dim inputs
+        y = torch.randn(3, 3, 4, dtype=torch.float)
+        onnx_testing.assert_onnx_program(onnx_program, args=(y,))
```

Comment:
Would be good to clean up some of these tests
```suggestion
        onnx_program = self.export(Model(), (torch.randn(2, 3, 4, dtype=torch.float),), dynamic_shapes=({0: torch.export.Dim("dim0")},))

        # different dim inputs
        onnx_testing.assert_onnx_program(onnx_program, args=(torch.randn(3, 3, 4, dtype=torch.float),))
```